clinica/iotools/converters/ixi_to_bids/ixi_to_bids_utils.py

```diff
@@ -0,0 +1,449 @@
+from pathlib import Path
+from typing import List, Optional, Union
+
+import nibabel as nib
+import pandas as pd
+from nilearn.image import concat_imgs
+
+from clinica.iotools.bids_utils import StudyName, bids_id_factory
+from clinica.utils.stream import cprint, log_and_raise
+
+__all__ = [
+    "read_clinical_data",
+    "define_participants",
+    "write_subject_data",
+    "write_sessions",
+    "write_scans",
+    "write_participants",
+    "check_modalities",
+]
+
+
+def _get_subjects_list_from_data(data_directory: Path) -> List[str]:
+    return list(
+        dict.fromkeys(
+            re.match(r"IXI\d{3}", path.name).group(0)
+            for path in data_directory.rglob(pattern="IXI*.nii.gz")
+            if re.match(r"IXI\d{3}", path.name)
+        )
+    )
+
+
+def _get_subjects_list_from_file(subjs_list_path: Path) -> List[str]:
+    return subjs_list_path.read_text().splitlines()
+
+
+def define_participants(
+    data_directory: Path,
+    subjs_list_path: Optional[Path] = None,
+) -> List[str]:
+    """
+    Defines the actual list of participants based on the (provided) subjects list filtered
+    using existing data.
+
+    Parameters
+    ----------
+    data_directory : Path to the raw data directory.
+    subjs_list_path : [Optional] Path to a text file containing a list of specific subjects to extract.
+
+    Returns
+    -------
+    The list of ids that were either present in the data directory or asked for with a specific text file, provided
+    associated images actually exists.
+
+    """
+
+    list_from_data = _get_subjects_list_from_data(data_directory)
+    if subjs_list_path is None:
+        return list_from_data
+    cprint("Loading a subjects list provided by the user...")
+    list_from_file = _get_subjects_list_from_file(subjs_list_path)
+    list_filtered = [subject for subject in list_from_file if subject in list_from_data]
+    invalid_subjects = list(set(list_from_file) - set(list_filtered))
+    if invalid_subjects:
+        cprint(
+            f"The subjects : {' , '.join(invalid_subjects)} do not have any associated data inside the directory {data_directory}"
+            f" and can not be converted."
+        )
+    return list_filtered
+
+
+def _rename_clinical_data_to_bids(column: str) -> str:
+    if column == "SEX_ID (1=m, 2=f)":
+        return "sex"
+    if column == "ETHNIC_ID":
+        return "ethnicity"
+    if column == "MARITAL_ID":
+        return "marital status"
+    if column == "OCCUPATION_ID":
+        return "occupation"
+    if column == "QUALIFICATION_ID":
+        return "qualification"
+    if column == "IXI_ID":
+        return "source_id"
+    if column == "DOB":
+        return "date of birth"
+    if column == "STUDY_DATE":
+        return "acq_time"
+    else:
+        return column.lower()
+
+
+def _get_sex_mapping() -> pd.DataFrame:
+    return pd.DataFrame({"SEX": ["male", "female"]}, index=[1, 2])["SEX"]
+
+
+class Clinical_Data_Mapping(str, Enum):
```

Comment:
Since this is a class:

```suggestion
class ClinicalDataMapping(str, Enum):
```